code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java

```diff
@@ -70,17 +72,14 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.concurrent.ConcurrentLinkedQueue;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
 
 public final class MessagingExtension extends Extension {
     private final static String SELF_TAG = "MessagingExtension";
 
-    private final ConcurrentLinkedQueue<Event> eventQueue;
     final InAppNotificationHandler inAppNotificationHandler;
     private boolean initialMessageFetchComplete = false;
     final LaunchRulesEngine messagingRulesEngine;
+    private final ExtensionApi extensionApi;
```

Comment:
do we need this?  can't we just call `getApi()` instead?

Reply:
i stored the extensionApi in a variable so it can be passed to MessagingUtils where the events are dispatched from. `getApi()` is only accessible in MessagingExtension.

Reply:
removed the extensionApi var and now use getApi() instead